src/EditorFeatures/Core/EditorConfigSettings/Updater/SettingsUpdateHelper.cs

```diff
@@ -87,7 +87,26 @@ internal static partial class SettingsUpdateHelper
                     optionValue = $"{optionValue}:{severity}";
                 }
 
-                var language = option.IsPerLanguage ? Language.CSharp | Language.VisualBasic : Language.CSharp;
+                Language language;
+                if (option is ISingleValuedOption singleValuedOption)
+                {
+                    language = singleValuedOption.LanguageName switch
+                    {
+                        LanguageNames.CSharp => Language.CSharp,
+                        LanguageNames.VisualBasic => Language.VisualBasic,
+                        null => Language.CSharp | Language.VisualBasic,
+                        _ => throw ExceptionUtilities.UnexpectedValue(singleValuedOption.LanguageName),
+                    };
+                }
+                else if (option is IPerLanguageValuedOption perLanguageValuedOption)
+                {
+                    language = Language.CSharp | Language.VisualBasic;
+                }
+                else
+                {
+                    throw ExceptionUtilities.UnexpectedValue(option);
```

Comment:
`IOption` is a public interface, so theoretically this path could be reachable _if_ a public entry point can supply this option. Do we know if the latter is possible?